### ISSUE TITLE
Panics when looking up metrics in carbonserver

### ISSUE THREAD

**Issue by jdblack** [go-graphite/go-carbon#343]
Version:  master (48fb01306384cd02fd98309c973151c35da7a47f)
Previous versions affected:  Yes
Issue:   Panics when looking up metrics in carbonserver
architecture: grafana + metricmakers  ->  carbon-c-relay & graphite-web 1.1.6 -> go-carbon

This  issue is targeted against  master, but I've seen this in multiple releases of go-carbon.  I'm sure it's a misconfiguration on our part, but I have -no- idea what it is.  In the past, we have dealt with this by disabling carbonserver and going back to go-carbon. However, we've suddently gotten a keen interest in using carbonserver because of https://github.com/lomik/go-carbon/pull/338.   We've tried a variety of options in carbonserver, and can't seem to get rid of the issue. our most recent configuration is:

```
# Generally we try turning most of the bool settings to true, but I tried turning all of htem off
to see if I could  identify what is causing the panic.
.....
[carbonserver]
# Please NOTE: carbonserver is not intended to fully replace graphite-web
# It acts as a "REMOTE_STORAGE" for graphite-web or carbonzipper/carbonapi
listen = ":8080"
enabled = false
buckets = 10
metrics-as-counters = false
read-timeout = "90s"
write-timeout = "90s"
query-cache-enabled = false
query-cache-size-mb = 1024
find-cache-enabled = false
trigram-index = false
scan-frequency = "5m0s"
cache-scan = true
max-globs = 100
fail-on-max-globs = false
trie-index = false
graphite-web-10-strict-mode = true
internal-stats-dir = ""
# Calculate /render request time percentiles for the bucket, '95' means calculate 95th Percentile. To disable this feature, leave the list blank
stats-percentiles = [99, 98, 95, 75, 50]
```

# A log snippet:
```
[2020-06-05T22:40:04.612Z] ERROR [access] fetch failed {"handler": "render", "url": "/render/?format=pickle&local=1&noCache=1&from=1591396504&until=1591396804&target=collectd.chef-frontend-ha.%2A.%2A.cpu-%2A.cpu-steal&now=1591396804", "peer": "ipaddressremoved:52132", "format": "pickle", "targets": ["collectd.chef-frontend-ha.*.*.cpu-*.cpu-steal"], "runtime_seconds": 0.000216782, "reason": "panic during serving the request", "stack": "github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).renderHandler.func1
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/render.go:199
runtime.gopanic
  /usr/local/Cellar/go/1.14.3/libexec/src/runtime/panic.go:969
runtime.panicmem
  /usr/local/Cellar/go/1.14.3/libexec/src/runtime/panic.go:212
runtime.sigpanic
  /usr/local/Cellar/go/1.14.3/libexec/src/runtime/signal_unix.go:695
github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).getExpandedGlobs
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/find.go:355
github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).prepareDataProto
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/render.go:343
github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).fetchWithCache
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/render.go:315
github.com/lomik/go-carbon/carbonserver.(*CarbonserverListener).renderHandler
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/render.go:207
net/http.HandlerFunc.ServeHTTP
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:2012
github.com/lomik/go-carbon/carbonserver.TraceHandler.func1
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/carbonserver/carbonserver.go:179
github.com/lomik/go-carbon/vendor/github.com/dgryski/httputil.TimeHandler.func1
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/vendor/github.com/dgryski/httputil/times.go:26
github.com/lomik/go-carbon/vendor/github.com/dgryski/httputil.TrackConnections.func1
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/vendor/github.com/dgryski/httputil/track.go:40
net/http.HandlerFunc.ServeHTTP
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:2012
net/http.(*ServeMux).ServeHTTP
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:2387
github.com/lomik/go-carbon/vendor/github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1
  /Users/jblackwell/go-carbon/_vendor/src/github.com/lomik/go-carbon/vendor/github.com/NYTimes/gziphandler/gzip.go:293
net/http.HandlerFunc.ServeHTTP
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:2012
net/http.serverHandler.ServeHTTP
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:2807
net/http.(*conn).serve
  /usr/local/Cellar/go/1.14.3/libexec/src/net/http/server.go:1895", "error": "runtime error: invalid memory address or nil pointer dereference", "http_code": 500}
```

**Comment by deniszh** [comment]
Hi @jdblack,
IIRC #338 will not work w/o trie or trigram index. Is default config for carbonserver still causing issues for you?
Default config - https://github.com/go-graphite/go-carbon#configuration (with enabled features, of course):
```ini
[carbonserver]
listen = "127.0.0.1:8080"
enabled = true
buckets = 10
metrics-as-counters = false
read-timeout = "60s"
write-timeout = "60s"
query-cache-enabled = true
query-cache-size-mb = 0
find-cache-enabled = true
trigram-index = true
scan-frequency = "5m0s"
trie-index = false
cache-scan = true
max-globs = 100
fail-on-max-globs = false
max-metrics-globbed  = 30000
max-metrics-rendered = 1000
graphite-web-10-strict-mode = true
internal-stats-dir = ""
stats-percentiles = [99, 98, 95, 75, 50]
```
You can set up separate go-carbon instance and check config changes there, without affecting main go-carbon.

**Comment by jdblack** [comment]
This does not happen in the released version of 0.15
